exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetBooleanPredicate.java

```diff
@@ -66,25 +83,38 @@ public boolean canDrop(RangeExprEvaluator<C> evaluator) {
       ExpressionPosition pos
   ) {
     return new ParquetBooleanPredicate<C>(name, args, pos) {
+      /**
+       * Evaluates a compound "OR" filter on the statistics of a RowGroup (the filter reads "filterA or filterB").
+       * Return value :<ul>
+       *   <li>NONE : only if all filters return NONE
+       *   <li>ALL : if one filter at least returns ALL
+       *   <li>SOME : all other cases
+       * </ul>
+       */
       @Override
-      public boolean canDrop(RangeExprEvaluator<C> evaluator) {
+      public RowsMatch matches(RangeExprEvaluator<C> evaluator) {
+        RowsMatch childMatch;
+        RowsMatch resultMatch = RowsMatch.NONE;
         for (LogicalExpression child : this) {
-          // "or" : as long as one branch is NOT ok to drop, we can NOT drop it.
-          if (!(child instanceof ParquetFilterPredicate) || !((ParquetFilterPredicate)child).canDrop(evaluator)) {
-            return false;
+          if (child instanceof ParquetFilterPredicate) {
+            childMatch = ((ParquetFilterPredicate) child).matches(evaluator);
+            if (childMatch != RowsMatch.NONE) {
```

Comment:
Check for `childMatch == RowsMatch.ALL` first.

Reply:
Done